lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -29,30 +30,35 @@ def initialize(options = {})
       end
 
       def purchase(amount, payment_method, options = {})
-        post = {}
-        add_auth_purchase_params(post, amount, payment_method, options)
         action = options[:operation_id] ? :purchase : :preauth
```

Comment:
why a purchase request could be converted into a preauth? 
What is the difference between authorize and preauth?

Reply:
This is because 3ds flow is required for every authorize and purchase transaction and the flow works according to the documentation [here](https://developer.nexi.it/en/modalita-di-integrazione/server-to-server/pagamento-2-steps) requiring to start with `init` (preauth) and then with `payment` (authorize and purchase)